network/network.go

```diff
@@ -916,11 +925,14 @@ func (n *network) attemptConnect(ip utils.IPDesc) error {
 // assumes the stateLock is not held. Returns an error if the peer's connection
 // wasn't able to be upgraded.
 func (n *network) upgrade(p *peer, upgrader Upgrader) error {
+	p.conn.SetReadDeadline(time.Now().Add(5 * time.Second))
```

Comment:
Do we need to set a write deadline here, too? I'm not that familiar with TLS upgrading.

Also, can we move `5 * time.Second` into a constant?

Reply:
it is hanging on the handshake when the person connecting has sent no data.  so we are just reading.
And the timeout is to ensure that we have some data within 5 seconds, or to hang up on them.
For an attack vector someone could just open a bunch of connections and end up eating up all our FD's just by sitting on them and not sending anything..